packages/core/src/internal/types.ts

```diff
@@ -281,26 +282,24 @@ export type ContractMetadata = {
   abiEvents: AbiEvents;
   abiFunctions: AbiFunctions;
   name: string;
-  network: Network;
+  chain: Chain;
 };
 export type AccountMetadata = {
   type: "account";
   name: string;
-  network: Network;
+  chain: Chain;
 };
 export type BlockMetadata = {
   type: "block";
   name: string;
-  network: Network;
+  chain: Chain;
 };
 
-// Network
+// Chain
 
-export type Network = {
-  name: string;
-  chainId: number;
-  transport: ReturnType<Transport>;
-  chain: Chain;
+export type Chain = {
+  chain: ViemChain;
```

Comment:
Nit, but I'd probably include `name: string` and `id: number` here as the chain name and ID that were provided directly by the user, and consider renaming `chain: ViemChain` to `viemChain: ViemChain`. The Viem chain feels secondary / "best effort".

This would change `args.chain.chain.id` -> `args.chain.id` in downstream files which makes more sense to me.